### PR TITLE
fix(frontend): do not render Retry nodes in UI DAG as they are virtual nodes. Fixes #4425 #2705

### LINKED ISSUE #4425: UI doesn't correctly return logs
### What steps did you take:
Made a simple KFP run on using an MNIST PyTorch training/evaluation script.

### What happened:
When making a KFP run, occasionally the UI does not properly retrieve logs, and gives the error `Failed to retrieve pod logs.` By inspecting the Chrome browser network setting, it appears to be an API call issue with the following error: `Could not get main container logs: Error: Unable to retrieve workflow status: [object Object].`.

### What did you expect to happen:
All the steps in the pipeline UI should correctly return logs.

### Environment:

How did you deploy Kubeflow Pipelines (KFP)?
Deployed KFP on Amazon EKS infrastructure.

KFP version: `v1.0.0`

KFP SDK version: `v1.0.0`

/kind bug

### PR DESCRIPTION
**Pull request by eterna2**
**Description of your changes:**

- Added `Retry` to list of virtual nodes to filter out when parsing DAG graph in the front-end.
- Added `retry-runtime` to mock server to visualize the change in rendered DAG.

The `Retry` node is a virtual node in Argo workflow (there are no corresponding physical pod for it, it is just used to internally monitor the actual pod to see if retry is required). It can be confusing to users if it is rendered in the UI (see #4425 #2705) as users will expect non-existence logs and other "physical" information to be returned when clicking on the node.

Before: Node 2 is the Retry node
image

After: Node 2 is removed from the DAG
image

**Checklist:**
- [x] The title for your pull request (PR) should follow our title convention. [Learn more about the pull request title convention used in this repository](https://github.com/kubeflow/pipelines/blob/master/CONTRIBUTING.md#pull-request-title-convention). 

   PR titles examples:
    * `fix(frontend): fixes empty page. Fixes #1234`
       Use `fix` to indicate that this PR fixes a bug.
    * `feat(backend): configurable service account. Fixes #1234, fixes #1235`
       Use `feat` to indicate that this PR adds a new feature. 
    * `chore: set up changelog generation tools`
       Use `chore` to indicate that this PR makes some changes that users don't need to know.
    * `test: fix CI failure. Part of #1234`
        Use `part of` to indicate that a PR is working on an issue, but shouldn't close the issue when merged.

- [ ] Do you want this pull request (PR) cherry-picked into the current release branch?
    
    If yes, use one of the following options:
  
    * **(Recommended.)** Ask the PR approver to add the `cherrypick-approved` label to this PR. The release manager adds this PR to the release branch in a batch update.
    *  After this PR is merged, create a cherry-pick PR to add these changes to the release branch. (For more information about creating a cherry-pick PR, see the [Kubeflow Pipelines release guide](https://github.com/kubeflow/pipelines/blob/master/RELEASE.md#option--git-cherry-pick).)

### CODE REVIEW

File: frontend/src/lib/WorkflowParser.ts [hunk 1]
```diff
@@ -408,7 +408,10 @@ export default class WorkflowParser {
   // meaningful from a user's perspective.
   public static isVirtual(node: NodeStatus): boolean {
     return (
-      (node.type === 'StepGroup' || node.type === 'DAG' || node.type === 'TaskGroup') &&
+      (node.type === 'StepGroup' ||
+        node.type === 'DAG' ||
+        node.type === 'TaskGroup' ||
+        node.type === 'Retry') &&
```
Comment (Bobgy) [inline frontend/src/lib/WorkflowParser.ts:414]: Can we add a unit test for the new change?

**Review by Bobgy — COMMENTED** [review]
Thank you so much @eterna2 !

/lgtm
The only ask is a small unit test case that covers Retry node.

File: frontend/src/lib/WorkflowParser.ts (hunk 1 above)
Comment (eterna2) [reply frontend/src/lib/WorkflowParser.ts:414]: unit test added

**Comment by Bobgy** [comment]
Awesome!
Thank you for the contribution!

/lgtm
/approve

**Pull request merged**